src/avram/credentials.cr

```diff
@@ -0,0 +1,125 @@
+class Avram::Credentials
+  getter url : String = ""
```

Comment:
Rather than storing the `url` I think it'd be great to make a `url` method that builds the URL from the other isntance vars. The reason for that is that I think it is a bit easier to follow and is less likely to have bugs. For example, if you did `Avram::Credentials.new("foobar").url` it would return `nil` right now, but if `url` was a method that built the URL it would work. It could probably be memoized but I imagine it isn't used much so may not be worth it. Up to you!

I think this could work almost as-is by renaming `build` to `url`

Reply:
ok, I think I see what you're saying. I'm gonna make a change and push that up to see if I'm on the right track.